### PR TITLE
feat: useSection

### PR DESCRIPTION
**Pull request by tlgimenes**

### CODE REVIEW

File: components/section.tsx [hunk 1]
```diff
@@ -182,7 +182,7 @@ export const withSection = <TProps,>(
             renderSalt,
           }}
         >
-          <Partial name={id} mode={partialMode}>
+          {/* <Partial name={id} mode={partialMode}> */}
```
Comment (lucis) [inline components/section.tsx:185]: 👀

File: runtime/fresh/routes/render.tsx [hunk 2]
```diff
@@ -106,11 +113,13 @@ export const handler = async (
   try {
     page = await ctx.state.resolve(
       { ...section, ...props },
-      { resolveChain },
+      resolveChain ? { resolveChain } : undefined,
       original,
     );
   } catch (err) {
     if (err instanceof HttpError) {
+      throw new Error("I Don't know what im doing");
```
Comment (mcandeia) [inline runtime/fresh/routes/render.tsx:121]: this is desirable? what is the use case? what happened?

**Review by mcandeia — APPROVED** [review]

File: components/section.tsx (hunk 1 above)
Comment (tlgimenes) [reply components/section.tsx]: done

File: runtime/fresh/routes/render.tsx (hunk 2 above)
Comment (tlgimenes) [reply runtime/fresh/routes/render.tsx]: done

**Pull request merged**